sdk/daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/crypto/Hash.scala

```diff
@@ -126,9 +130,16 @@ object Hash {
       add(a.length).add(a)
     }
 
+    private def add(fieldNumber: Int, v: String): this.type =
```

Comment:
This massive overloading makes it hard to see when default values are simply omitted. I'd recommend to make this explicit in the names of the new methods, such as 
```suggestion
    private def addUnlessDefault(fieldNumber: Int, v: String): this.type =
```